std/encoding.d

```diff
@@ -652,7 +662,8 @@ template EncoderInstance(E)
 
 private template GenericEncoder()
 {
-    bool canEncode(dchar c)
+    @safe:
```

Comment:
This seems a bit iffy to me. Since this is a template, shouldn't the compiler already be inferring `@safe`? If not, we should identify which parts aren't, and figure out why. (From the looks of it, it may just be that one function that needed the `@trusted`.)

A `@safe` unittest that instantiates this template should be enough to ensure that the code doesn't accidentally become un-`@safe` again.